common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.event.ChatMessageReceivedEvent;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class TradeMarketAutoOpenChatFeature extends UserFeature {
+
+    // Type the price in emeralds or type 'cancel' to cancel:
+    // Type the amount you wish to buy or type 'cancel' to cancel:
+    // Type the item name or type 'cancel' to cancel:
+    private static final Pattern TYPE_TO_CHAT_PATTERN = Pattern.compile("§6Type the .* or type 'cancel' to cancel:");
+
+    private boolean openChatWhenContainerClosed = false;
+
+    @SubscribeEvent
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!WynnUtils.onWorld()) return;
+
+        if (TYPE_TO_CHAT_PATTERN.matcher(event.getCodedMessage()).matches()) {
+            openChatWhenContainerClosed = true;
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerClose(ScreenClosedEvent event) {
```

Comment:
do we use this same logic in legacy? just wondering if it's at all possible for the container to close before the message is sent

Reply:
I didn't check until I wrote this, but yes, same logic.